core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -715,7 +715,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
   }
 
-  private def mergeApplicationListing(
+  private[history] def mergeApplicationListing(
```

Comment:
Instead of changing visibility, could you use `org.scalatest.PrivateMethodTester` and `PrivateMethod` in the test suite?